### PR TITLE
[5.3] app:name - add file exists check for ModelFactory.php during namespace replacement

### PR DESCRIPTION
**Pull request by ralphschindler**
Fixes issue #16575
- add file exists check to ModelFactory.php file when performing app:name command

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
The `app:name` is only ever meant to be run on a new application tbh.

**Comment by ralphschindler** [comment]
Agreed, but in the case someone setting up a new app happens to remove the `ModelFactory.php` then run app:name, it will leave the new application in a broken state because app:name did not have a chance to dump the autoload, and there is no indication to the developer that is the thing that fixes a broken app.

I think there are some legitimate use cases where someone might have trashed `database/` directory or even perhaps switched to phinx, for example.  If they started building migrations before touching `app/` code (where you expect the namespace to be changed) then they notice they want to change the namespace and run the command.

All that said, we're really talking about `AppNameCommand` doing even the smallest bit of due diligence in determining if the file its trying to update is there.  I think this one liner has loads of benefits for nearly zero drawbacks.

File: src/Illuminate/Foundation/Console/AppNameCommand.php [hunk 1]
```diff
@@ -223,9 +223,11 @@ protected function setServicesConfigNamespace()
      */
     protected function setDatabaseFactoryNamespaces()
     {
-        $this->replaceIn(
-            $this->laravel->databasePath().'/factories/ModelFactory.php', $this->currentRoot, $this->argument('name')
-        );
+        $modelFactoryFile = $this->laravel->databasePath().'/factories/ModelFactory.php';
+
+        if ($this->files->exists($modelFactoryFile)) {
+            $this->replaceIn($modelFactoryFile, $this->currentRoot, $this->argument('name'));
```
Comment (lucasmichot) [inline src/Illuminate/Foundation/Console/AppNameCommand.php]: Would be simpler to do that check directly in `replaceIn` function

Comment (ralphschindler) [inline src/Illuminate/Foundation/Console/AppNameCommand.php]: Yes, that is a good idea, ... updating PR.

**Comment by browner12** [comment]
i'm curious, are there reasons that people change their App name, other than for purely cosmetic reasons?  when this feature first appeared I used it on a new app, but quickly found it to be a little bit of a PITA while providing no real benefit.

if it is purely cosmetic, I would almost argue that we get rid of it.

**Comment by taylorotwell** [comment]
I agree renaming App is generally not something I do or even recommend, but while the command is here might as well make it work I guess.

I would be open to removing it or even extracting the whole functionality into a package that is maintained separately as it is a little bit of a time sink. If anyone wants to take that on ping me!

**Comment by ralphschindler** [comment]
Thanks for merging @taylorotwell.

If in the future there needs to be a discussion on keeping/removing it, I can add to the argument there - we (at work) find it beneficial for a host of reasons.

**Comment by browner12** [comment]
hey @ralphschindler, would you mind listing some of the reasons here for documentation?

I'm also personally curious to hear what they are.

Thanks!

**Pull request merged**